test/e2e/providers/vsphere/vsphere.go

```diff
@@ -133,9 +143,320 @@ func (p *Provider) GetType() config.PlatformType {
+	if err := p.ensureFSSConfigMap(client); err != nil {
+		return err
+	}
+	return p.ensureWindowsCSIDaemonset(client)
+}
+
+// ensureFSSConfigMap creates a feature state switch ConfigMap for Windows Nodes. The FSS used by Linux nodes is
+// unusable as it does not set csi-windows-support to true
+func (p *Provider) ensureFSSConfigMap(client client.Interface) error {
+	fssCM := core.ConfigMap{
+		ObjectMeta: meta.ObjectMeta{
+			Name: windowsFSSName,
+		},
+		Data: map[string]string{
+			"block-volume-snapshot":            "true",
+			"cnsmgr-suspend-create-volume":     "true",
+			"csi-auth-check":                   "true",
+			"csi-migration":                    "true",
+			"improved-csi-idempotency":         "true",
+			"improved-volume-topology":         "false",
+			"online-volume-extend":             "true",
+			"topology-preferential-datastores": "true",
+			"csi-windows-support":              "true",
+		},
+		BinaryData: nil,
+	}
+
+	// See if the ConfigMap already exists in the state we expect it to be in.
+	existingCM, err := client.CoreV1().ConfigMaps(csiNamespace).Get(context.TODO(), windowsFSSName, meta.GetOptions{})
+	if err != nil {
+		if !k8sapierrors.IsNotFound(err) {
+			return fmt.Errorf("error getting existing FSS ConfigMap: %w", err)
+		}
+		if reflect.DeepEqual(existingCM.Data, fssCM.Data) {
+			// ConfigMap is already as expected, nothing to do here.
+			return nil
+		}
+		// Delete the ConfigMap as it has the wrong data.
+		err = client.CoreV1().ConfigMaps(csiNamespace).Delete(context.TODO(), windowsFSSName, meta.DeleteOptions{})
+		if err != nil {
+			return fmt.Errorf("error deleting existing FSS ConfigMap: %w", err)
+
+		}
+	}
+	_, err = client.CoreV1().ConfigMaps(csiNamespace).Create(context.TODO(), &fssCM, meta.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("could not create FSS ConfigMap: %w", err)
+	}
+	return nil
+}
+
+// ensureWindowsCSIDaemonSet deploys the Windows CSI driver DaemonSet if it doesn't already exist
+func (p *Provider) ensureWindowsCSIDaemonset(client client.Interface) error {
+	dsName := "vmware-vsphere-csi-driver-node-windows"
+	directoryType := core.HostPathDirectory
+	directoryOrCreate := core.HostPathDirectoryOrCreate
+	ds := apps.DaemonSet{
+		ObjectMeta: meta.ObjectMeta{
+			Name: dsName,
+		},
+		Spec: apps.DaemonSetSpec{
+			Selector: &meta.LabelSelector{MatchLabels: map[string]string{"app": dsName}},
+			Template: core.PodTemplateSpec{
+				ObjectMeta: meta.ObjectMeta{
+					Labels: map[string]string{"app": dsName},
+				},
+				Spec: core.PodSpec{
+					PriorityClassName:  "system-node-critical",
+					NodeSelector:       map[string]string{core.LabelOSStable: "windows"},
+					ServiceAccountName: "vmware-vsphere-csi-driver-node-sa",
+					OS:                 &core.PodOS{Name: core.Windows},
+					Tolerations: []core.Toleration{
+						{
+							Key:    "os",
+							Value:  "Windows",
+							Effect: core.TaintEffectNoSchedule,
+						},
```

Comment:
We should make this a constant or a global variable that we use everywhere in the e2e tests. Can be done in a follow up PR.